### PR TITLE
[AN-5001][AN-5002][AN-5003][AN-5004][AN-5005][AN-5006] Text search

### PR DESCRIPTION
**Pull request by JPonte**
#### APK
[Download build #8489](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8489/artifact/build/artifact/wire-dev-PR608-8489.apk)
[Download build #8491](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8491/artifact/build/artifact/wire-dev-PR608-8491.apk)

### CODE REVIEW

**Comment by JPonte** [comment]
build again

**Comment by xuanzhaopeng** [comment]
bug C426653 Verify conversation scrolls to the selected result when searching from conversation

**Comment by JPonte** [comment]
build again

File: build.gradle [hunk 1]
```diff
@@ -58,7 +58,7 @@ ext {
     majorVersion = "2.28."
 
     compileSdkVersion = 24
-    minSdkVersion = 17
```
Comment (hakonbo) [inline build.gradle]: This won't fly with a bunch of our users... ;)

File: app/src/main/scala/com/waz/zclient/conversation/CollectionFragment.scala [hunk 2]
```diff
@@ -157,10 +291,22 @@ class CollectionFragment extends BaseFragment[CollectionFragment.Container] with
...
   }
+
+  override def onStart(): Unit = {
+    super.onStart()
+    getActivity.getWindow.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN)
+  }
+
+  override def onStop(): Unit = {
+    super.onStop()
+    getActivity.getWindow.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE|WindowManager.LayoutParams.SOFT_INPUT_STATE_HIDDEN)
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/conversation/CollectionFragment.scala]: There is a possibility to set this via `getGlobalLayoutController`, can't remember the details, but might be worth having a look at that one...

File: app/src/main/scala/com/waz/zclient/conversation/CollectionFragment.scala [hunk 3]
```diff
@@ -86,43 +122,140 @@ class CollectionFragment extends BaseFragment[CollectionFragment.Container] with
...
+      override def beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int): Unit = {}
+
+      override def onTextChanged(s: CharSequence, start: Int, before: Int, count: Int): Unit = {
+        if (s.toString.trim.length() <= 1) {
+          controller.contentSearchQuery ! ContentSearchQuery.empty
+        } else {
+          controller.contentSearchQuery ! ContentSearchQuery(s.toString)
+        }
+        if (s.toString.nonEmpty){
+          searchBoxClose.setVisibility(View.VISIBLE)
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/conversation/CollectionFragment.scala]: Maybe use `setVisible` in `com.waz.zclient.util` to minimize this a bit.

File: app/src/main/scala/com/waz/zclient/conversation/CollectionFragment.scala [hunk 4]
```diff
@@ -86,43 +122,140 @@ class CollectionFragment extends BaseFragment[CollectionFragment.Container] with
...
+      override def onKey(v: View, keyCode: Int, event: KeyEvent): Boolean = {
+        if (event.getAction == KeyEvent.ACTION_DOWN && keyCode == KeyEvent.KEYCODE_BACK) {
+          v.clearFocus()
+        }
+        false
+      }
+    })
+    searchBoxView.setOnFocusChangeListener(new OnFocusChangeListener {
+      override def onFocusChange(v: View, hasFocus: Boolean): Unit = {
+        if (!hasFocus && searchBoxView.getText.length() == 0) {
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/conversation/CollectionFragment.scala]: Maybe use `setVisible` in `com.waz.zclient.util` to minimize this a bit.

File: app/src/main/res/layout/fragment_collection.xml [hunk 5]
```diff
@@ -66,24 +66,102 @@
...
+            android:descendantFocusability="beforeDescendants"
+            android:focusableInTouchMode="true"
+            >
+
+            <com.waz.zclient.ui.text.TypefaceEditText
+                android:id="@+id/search_box"
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:textColor="?wirePrimaryTextColor"
+                android:paddingLeft="@dimen/wire__padding__20"
```
Comment (hakonbo) [inline app/src/main/res/layout/fragment_collection.xml]: Not really a big fan of these `wire__padding__XX`. This has been done hysterically/historically to get around the lint check for values in XML. Perhaps you can find a way to describe this padding a bit better in a new variable? Or use a style? Or if not, maybe just use `20dp` directly and add an ignore tag to this element?

File: build.gradle (hunk 1 above)
Comment (JPonte) [inline build.gradle]: oops, did that to test something and forgot about it...

File: app/src/main/scala/com/waz/zclient/conversation/CollectionFragment.scala (hunk 2 above)
Comment (JPonte) [inline app/src/main/scala/com/waz/zclient/conversation/CollectionFragment.scala]: ah you're right. setSoftInputModeForPage. thanks!

**Comment by red-avtovo** [comment]
searching is making from the start of the word, but highlighting of the results - whatever it matches

**Comment by red-avtovo** [comment]
searching for multiple words joins them with AND, but highlight - OR

**Pull request merged**